src/manager.ts

```diff
@@ -184,16 +208,122 @@ export class SecretsManager implements ISecretsManager {
     this._attachedInputs.delete(attachedId);
   }
 
-  private _connector: ISecretsConnector;
+  private _check(token: symbol, namespace: string): void {
+    if (Private.isLocked() || Private.namespace.get(token) !== namespace) {
+      throw new Error(
+        `The secrets namespace ${namespace} is not available with the provided token`
+      );
+    }
+  }
+
   private _attachedInputs = new Map<string, HTMLInputElement>();
   private _ready: PromiseDelegate<void>;
 }
 
+/**
```

Comment:
```suggestion
}

Object.freeze(SecretsManager);
Object.freeze(SecretsManager.prototype);

/**
```

re: the monkey-patching attack @trungleduc discussed

Reply:
I tested this with [a toy example](https://www.typescriptlang.org/play/?#code/MYGwhgzhAEAKBOB7ALgU2GgJtA3gKGmkgE8A7YaAc1WQAoBLTALmgmXntMoEpcDDo8GgFd4pOEgC29CKgB0QiIhAA3VLQAG1ZAFpFwkMiYASHIwC+G7gG5+5vPbwB5AEYArdMjkAzIagBe6ggonqiYcgAOSMgoxBGoNnh4KoiM0LQk5Om8ALwAfHyEwIikbNCcbGDkqNA50KSoAO4SIRhhtIlFJWXateWlyFXA8tq2hAwDQzWQ0MKkANakiI3iM1XE3HK9dQzMrOycPLUF+ALQxaXK8iCIlLQA5ACyAJIAKo-3ADTlmJ0CQshROJtLs-uYxudulc5Dc7mBGmB6Mh+pVqlsaA80Gx7txEuZuB1bHggA).